algebird-core/src/main/scala/com/twitter/algebird/CorrelationGroup.scala

```diff
@@ -0,0 +1,107 @@
+package com.twitter.algebird
+
+object Correlation {
+  def apply(x: (Double, Double)): Correlation =
+    Correlation(c2 = 0, m2Left = 0, m2Right = 0, m1Left = x._1, m1Right = x._2, 1L)
+
+  implicit val group: Group[Correlation] = CorrelationGroup
+}
+
+/**
+ * A class to calculate covariance and the first two central moments of a sequence of pairs of Doubles, from which the
+ * pearson correlation coeifficient can be calculated.
+ *
+ * m{i}Left denotes the ith central moment of the first projection of the pair.
+ * m{i}Right denotes the ith central moment of the second projection of the pair.
+ * c2 the covariance equivalent of the second central moment, i.e. c2 = Sum_(l,r) (l - m1Left)*(r - m1Right).
+ *
+ */
+case class Correlation(
+    c2: Double,
+    m2Left: Double,
+    m2Right: Double,
+    m1Left: Double,
+    m1Right: Double,
+    m0: Long
```

Comment:
Can we make this a `Double` actually? If we make this a double, we can actually make Correlation a `VectorSpace` which allows us to do decayed values on Correlation (which is a pretty cool idea)....

I've been using decayed 2nd moments in a project and it seems to be a useful feature transformation since it can give you decayed z-scores.

Reply:
Taking a closer look at section 3.1 of the paper I linked to in the PR description, it looks like the calculation for weights really is as simple as making this a `Double` (the choice of notation in the two papers is annoyingly different though). Given how easy this is, it seems like a no-brainer to implement. Good call-out here.

Section 4.1 talks about weighted correlation being used as a metric (seems like an easy enough method to add) and 4.2 exponentially weighted moving correlation (which seems like something that should live outside of this particular abstraction--do you agree?).